File: backend/app/models/product.py
from decimal import Decimal
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import Index, Numeric, String, Float, Boolean, DateTime
from sqlalchemy.sql import func
from database.base import Base

class Product(Base):
    __tablename__ = "products"

    id: Mapped[int] = mapped_column(primary_key=True)
    sku: Mapped[str] = mapped_column(String, unique=True, nullable=False)
    name: Mapped[str] = mapped_column(String, nullable=False)
    category: Mapped[str | None] = mapped_column(String)
    unit_price: Mapped[Decimal] = mapped_column(
        Numeric(12, 2),
        nullable=False
    )
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)
    created_at: Mapped[DateTime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

    __table_args__ = (
        Index("idx_products_sku", "sku"),
    )

File: frontend/src/types/product.ts
export interface ProductResponse {
  id: number
  sku: string
  name: string
  category: string | null
  unit_price: string
  is_active: boolean
}

export interface CreateProductRequest {
  sku: string
  name: string
  category?: string | null
  unit_price: number
}

export interface UpdateProductRequest {
  name?: string
  category?: string | null
  unit_price?: number
  is_active?: boolean
}
